### ISSUE TITLE
Look at self-hosted CI to improve selenium reliability

### ISSUE THREAD

**Issue by bovine3dom** [tridactyl/tridactyl#2256]
We can afford a fairly beefy server if we need it. 

Have seen https://github.com/drone/drone , https://concourse-ci.org/ and https://buddy.works/ recommended

**Comment by glacambre** [comment]
I don't think this is necessary. I managed to make Firenvim's selenium testsuite fairly reliable by removing as many `sleep` calls as possible and by removing as many dependencies on external websites as possible (now everything is done on `file:///` pages I think). It should be possible to do approximately the same thing with Tridactyl I think.

**Comment by bovine3dom** [comment]
That's a good shout - we can test `*open` by running a local web server (e.g. caddy) to avoid the file:// limitations.
